src/shared/ui/list/list.tsx

```diff
@@ -0,0 +1,44 @@
+// todo delete TextWithLink
+import classNames from 'classnames/bind';
+import { TextWithLink } from '../text-with-link';
+import { LinkList } from '@/widgets/required/required.types';
+
+import styles from './list.module.scss';
+
+export type ListProps = {
+  data: ListData;
+  marked?: boolean;
+};
+
+export type ListData = (string | LinkList)[] | [] | undefined;
```

Comment:
This is the type that `TextWithLink` component works with, changing it will break the code. Such changes are possible only after refactoring `TextWithLink` component, it's out of scope this task

Reply:
I see...```TextWithLink``` has confusing name and looks more like some kind of helper function....
ok